### ISSUE TITLE
How to build from master on macOS Sierra 10.12.6?

### ISSUE THREAD

**Issue by ElhemEnohpi** [mpv-player/mpv#6774]
### mpv version and platform
git master, macOS 10.12.6, Xcode 9.2

### Reproduction steps
Sorry if I'm asking the wrong thing, in the wrong place, I'm not very familiar with all this. I just want to get a recent version of mpv, since I'm having some troubles with the last release from October.

I  have Xcode 9.2 installed, I installed homebrew, and did `brew install --HEAD mpv`.

### Expected behavior
Up-to-date mpv is installed.
### Actual behavior
Errors.
### Log file
This is the log file from the last time I tried it:
https://0x0.st/z26d.python3

**Comment by Akemi** [comment]
will be fixed with #6703. one problem with "cocoa-cb: fix optional cases on macOS 10.12" the other with "cocoa-cb: conditional compilation for Dark Mode and Material features"

**Comment by ElhemEnohpi** [comment]
Thanks @Akemi - unfortunately I don't know enough to apply that patch myself. Is there some other way I could obtain a recent build, maybe compile your fork? Or would I have to wait for the pull requests to be merged or until I'm able to update my OS to Mojave?

**Comment by Akemi** [comment]
`brew edit mpv` and change that line `head "https://github.com/mpv-player/mpv.git"` to `head "https://github.com/Akemi/mpv.git", :branch => 'swift5'.`

alternative add a line to the mpv formula like this

```
patch do
      url "https://patch-diff.githubusercontent.com/raw/mpv-player/mpv/pull/6703.patch"
end
```

**Comment by ElhemEnohpi** [comment]
Thanks... I tried that but it looks like it didn't succeed. I think there's a problem getting the SDK version? There's an error in ~/Library/Logs/Homebrew/mpv/02.python3: 
```
Checking for macOS SDK  : xcrun: error: unexpected SDK settings, missing value for 'ProductBuildVersion'
xcrun: error: unable to lookup item 'ProductBuildVersion' in SDK '/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk'
```

Doing `xcrun -sdk macosx --show-sdk-version` gives `10.13`, but doing `xcrun -sdk macosx --show-sdk-build-version` gives the error above.

If I look in System Information, it says
```
Developer Tools:
  Version:	9.2 (9C40b)
  Location:	/Volumes/ExternalHD/Applications/Xcode.app
  Applications:
  Xcode:	9.2 (13772)
  Instruments:	9.2 (63753)
  SDKs:
  iOS:
  11.2:	(15C107)
  iOS Simulator:
  11.2:	(15C107)
  macOS:
  10.13:	(17C76)
```

The output of `brew install --HEAD mpv` shows:
``` 
File "/private/tmp/mpv-20190707-49679-5hj0zb/.waf3-2.0.9-10a533182bd85c3f45a157fb5d62db50/waflib/Configure.py", line 85, in execute
    super(ConfigurationContext,self).execute()
  File "/private/tmp/mpv-20190707-49679-5hj0zb/.waf3-2.0.9-10a533182bd85c3f45a157fb5d62db50/waflib/Context.py", line 85, in execute
    self.recurse([os.path.dirname(g_module.root_path)])
  File "/private/tmp/mpv-20190707-49679-5hj0zb/.waf3-2.0.9-10a533182bd85c3f45a157fb5d62db50/waflib/Context.py", line 126, in recurse
    user_function(self)
  File "/private/tmp/mpv-20190707-49679-5hj0zb/wscript", line 1047, in configure
    ctx.load('detections.compiler_swift')
  File "/private/tmp/mpv-20190707-49679-5hj0zb/.waf3-2.0.9-10a533182bd85c3f45a157fb5d62db50/waflib/Configure.py", line 155, in load
    func(self)
  File "/private/tmp/mpv-20190707-49679-5hj0zb/waftools/detections/compiler_swift.py", line 170, in configure
    __find_macos_sdk(ctx)
  File "/private/tmp/mpv-20190707-49679-5hj0zb/waftools/detections/compiler_swift.py", line 144, in __find_macos_sdk
    ctx.end_msg(sdk + ' (version found: ' + ctx.env.MACOS_SDK_VERSION + ')')
TypeError: can only concatenate str (not "list") to str

READ THIS: https://docs.brew.sh/Troubleshooting
```
After reading some other issues, I tried to do:
```
export MACOS_SDK_VERSION="10.13.2"
brew install --HEAD mpv
```
but that didn't work either... not sure what to try next.

**Comment by Akemi** [comment]
set your toolchain path to your xcode.app not the command line tools for now, `sudo xcode-select -s /Volumes/ExternalHD/Applications/Xcode.app`. i will look into the problem with the command line tools when i got time. seems like it doesn't support an sdk build version, which is kinda shit.

**Comment by ElhemEnohpi** [comment]
Ok, after changing to Xcode.app, `xcrun -sdk macosx --show-sdk-build-version` now gives `17C76`, and I was able to compile your fork, thanks!

**Comment by Akemi** [comment]
i fixed that command line tools problem in my #6703 PR now.

**Comment by Akemi** [comment]
fixed with #6703.
